src/main/java/user/database/UserFileGateway.java

```diff
@@ -0,0 +1,38 @@
+package user.database;
+
+import user.entities.User;
+import user.entities.UserFactory;
```

Comment:
is it ok to have gateways dependent on entities? (I'm not sure since the example repo has presenters in the abr package)